File: MATEMÁTICA/1554/main.js
function getDistance(x1, y1, x2, y2) {
    return Math.sqrt(Math.pow(x1 - x2, 2) + Math.pow(y1 - y2, 2));
}

function nearest(n, balls) {
    let minDistance = getDistance(balls[0][0], balls[0][1], 1420, 2840);
    let nearestBall = 0;
    for (let ball = 1; ball < n + 1; ball++) {
        let distance = getDistance(balls[0][0], balls[0][1], balls[ball][0], balls[ball][1]);
        if (distance < minDistance) {
            minDistance = distance;
            nearestBall = ball;
        }
    }
    return nearestBall;
}

let lines = require('fs').readFileSync('/dev/stdin', 'utf-8').split('\n');
let c = parseInt(lines.shift());
while (c--) {
    let n = parseInt(lines.shift());
    let balls = [];
    for (let i = 0; i < n + 1; i++) balls.push(lines.shift().split(' ').map(Number));
    console.log(nearest(n, balls));
}

File: MATEMÁTICA/1554/main.dart
import 'dart:io';
import 'dart:math';

double getDistance(int x1, int y1, int x2, int y2) {
  return sqrt(pow(x1 - x2, 2) + pow(y1 - y2, 2));
}

int nearest(int n, List<List<int>> balls) {
  double minDistance = getDistance(balls[0][0], balls[0][1], 1420, 2840);
  int nearestBall = 0;
  for (int ball = 1; ball < n + 1; ball++) {
    double distance = getDistance(balls[0][0], balls[0][1], balls[ball][0], balls[ball][1]);
    if (distance < minDistance) {
      minDistance = distance;
      nearestBall = ball;
    }
  }
  return nearestBall;
}

void main() {
  int c = int.parse(stdin.readLineSync()!);
  while (c-- > 0) {
    int n = int.parse(stdin.readLineSync()!);
    List<List<int>> balls = [];
    for (int i = 0; i < n + 1; i++) balls.add(stdin.readLineSync()!.split(' ').map(int.parse).toList());
    print(nearest(n, balls));
  }
}
